pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -58,29 +58,105 @@ const (
 	endpointQueryData      = "queryData"
 )
 
+type errorSource string
+
+const (
+	grafanaSource          errorSource = "grafana"
+	userSource             errorSource = "user"
+	downstreamClientSource errorSource = "downstreamClient"
+	downstreamServerSource errorSource = "downstreamServer"
```

Comment:
I don't understand the use case of having these two divided.

Reply:
"downstream" is when downstream return anything else than http response code 200.
I created this two to make the differences between the "4xx" errors, which are most likely due to the user (client), like auth, and the "5xx" errors, which should never happen and are real downstream error

I don't have any strong opinion on this, what do you suggest?